### ISSUE TITLE
[iOS] Trouble installing: One scheme/target working but the other doesn't?

### ISSUE THREAD

**Issue by dwilt** [react-native-config/react-native-config#75]
I followed your steps and have 2 schemes and targets:

`GreatJonesStreet [Dev]` and `GreatJonesStreet [Prod]`

I went through the entire installation process and when I build with the Dev target/scheme, everything works perfectly. It's pulling in my variables and all is dandy. I switch to the Prod target, and try and build, and get this:

![image](https://cloud.githubusercontent.com/assets/1065539/22102073/f2acd0f2-ddea-11e6-929c-e940a5fe0943.png)

Any ideas?

`react-native-config: 0.2.1`
`react-native: 0.37.0`
`react: 15.3.2`
`Xcode: 8.2.1`
`node: 6.9.1`

**Comment by hobofan** [comment]
I doubt that it's the solution for you issue, but I had a similar issue where react-native-config worked with one build configuration but not with another custom created one.

There is this block in the README:

> ### Extra step for iOS to support Info.plist
> 
> * Go to your project -> Build Settings -> All
> * Search for "preprocess" 
> * Set `Preprocess Info.plist File` to `Yes`
> * Set `Info.plist Preprocessor Prefix File` to `${CONFIGURATION_BUILD_DIR}/GeneratedInfoPlistDotEnv.h`
> * Set `Info.plist Other Preprocessor Flags` to `-traditional`
> * If you don't see those settings, verify that "All" is selected at the top (instead of "Basic")

The `${CONFIGURATION_BUILD_DIR}/GeneratedInfoPlistDotEnv.h` part assumes that the `CONFIGURATION_BUILD_DIR` of main project is the same as the one of the `ReactNativeConfig.xcodeproj`. This only works if you use one of two default `Debug` and `Release` build configurations.

**Comment by hobofan** [comment]
Something more related to your actual issue 😄 : I looked through your video from https://github.com/luggit/react-native-config/issues/76#issuecomment-273851265, and one thing you didn't replicate across targets/schemes (or at least didn't mention) was the part where you add the Pre-Action script to the scheme. That's the best guess I have, everything else looks good.

**Comment by dwilt** [comment]
@hobofan thanks for the responses. Isn't that what I'm mentioning here? https://youtu.be/VtrumM39wEg?t=5m7s

**Comment by hobofan** [comment]
Yes, but towards the end you showed that you "configured everything the same way" and just showed the build settings. You also have to make sure to add the Pre-Action script to every scheme you are using. Just trying to be thorough here ;)

**Comment by dwilt** [comment]
Ahhh, I see what you mean. Yes, I had it in there and forgot to show it. Thanks for making that clear. I usually try to be very clear/verbose to make sure no one gets confused.

**Comment by Balasnest** [comment]
I have 2 schemes and targets. Development and Production. When I run dev is working fine but for other schema target, getting the same **development config object** dev env file. Not related to production env file. 

why is this happening?
